src/components/smartSearch/filters/MostActive.tsx

```diff
@@ -1,62 +1,60 @@
+import { FormEvent } from 'react';
 import { FormattedMessage as Msg } from 'react-intl';
 import { Box, Button, Divider, MenuItem, Typography } from '@material-ui/core';
-import { FormEvent, useState } from 'react';
 
-import { isFilterWithId } from '../utils';
 import StyledNumberInput from '../inputs/StyledNumberInput';
 import StyledSelect from '../inputs/StyledSelect';
 import TimeFrame from './TimeFrame';
-import { ZetkinSmartSearchFilter } from 'types/zetkin';
-import { ZetkinSmartSearchFilterWithId } from 'types/smartSearch';
+import useSmartSearchFilter from 'hooks/useSmartSearchFilter';
+import { MostActiveFilterConfig, NewSmartSearchFilter, OPERATION, SmartSearchFilterWithId, ZetkinSmartSearchFilter } from 'types/smartSearch';
 
 interface MostActiveProps {
-    filter: ZetkinSmartSearchFilterWithId | ZetkinSmartSearchFilter | null;
-    onSubmit: (filter: ZetkinSmartSearchFilterWithId | ZetkinSmartSearchFilter) => void;
+    filter:  SmartSearchFilterWithId<MostActiveFilterConfig> |  NewSmartSearchFilter ;
```

Comment:
Ok, I see it being used here -- what's the advantage of having this type rather than allowing the component to accept an optional filter?

Reply:
@richardolsson reccomended setting `selectedFilter` to `{ type: FILTER_TYPE }` in order to handle creating new filters. This means, the possible value of `selectedFilter` counld be `{ type: FILTER_TYPE }`, or it could be `SmartSearchFilterWithId`, depending if creating a new filter or modifying an existing one. I typed it because this data structure will be used across components, so it's clearer in the component `MostActive` that it's possible you will get an object with only the `type` property :)

Reply:
This typing is really useful, because right now, there's always an assumption with the `useSmartSearchFilter` hook and the `MostActive` component that the value of filter will be either an existing or a new filter. If in the future, we change the way the new filter object looks, for example we don't pass in the type, VS code and eslint will show errors saying "hey we expect a `type`" which is good, because it does require a type to work. It will make refactoring and changing easier in the future, and debugging easier now.

Reply:
I can't remember saying this anywhere, but I do remember thinking it. So either I don't know what I'm saying or @djbusstop can read my thoughts. 😬 

But yeah, anyway, my thought was that instead of having a separate _type_ variable and using the absence of a filter (or just the ID) as the indicator of whether a filter is new or being edited, we could solve both of those by doing more or less exactly what @djbusstop has done here.